static/app/views/settings/organizationMembers/inviteRequestRow.tsx

```diff
@@ -78,33 +81,49 @@ const InviteRequestRow = ({
         )}
       </div>
 
-      <StyledRoleSelectControl
-        name="role"
-        disableUnallowed
-        onChange={r => onUpdate({role: r.value})}
-        value={inviteRequest.role}
-        roles={allRoles}
-      />
-
-      <TeamSelectControl
-        name="teams"
-        placeholder={t('Add to teams\u2026')}
-        onChange={(teams: OnChangeArgs) =>
-          onUpdate({teams: (teams || []).map(team => team.value)})
-        }
-        value={inviteRequest.teams}
-        options={allTeams.map(({slug}) => ({
-          value: slug,
-          label: `#${slug}`,
-        }))}
-        clearable
-      />
+      {canApprove ? (
+        <StyledRoleSelectControl
+          name="role"
+          disableUnallowed
+          onChange={r => onUpdate({role: r.value})}
+          value={inviteRequest.role}
+          roles={allRoles}
+        />
+      ) : (
+        <div>{inviteRequest.roleName}</div>
+      )}
+      {canApprove ? (
+        <TeamSelectControl
+          name="teams"
+          placeholder={t('Add to teams\u2026')}
+          onChange={(teams: OnChangeArgs) =>
+            onUpdate({teams: (teams || []).map(team => team.value)})
+          }
+          value={inviteRequest.teams}
+          options={allTeams.map(({slug}) => ({
+            value: slug,
+            label: `#${slug}`,
+          }))}
+          clearable
+        />
+      ) : (
+        <div>{inviteRequest.teams.join(', ')}</div>
+      )}
 
       <ButtonGroup>
+        <Button
```

Comment:
@Neo-Zhixing I think we probably want to make sure the deny button also has a tooltip when it's disabled and you hover over it.

Reply:
Sure - tooltip added.